components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4226,15 +4228,55 @@ public static String getJwkThumbPrint(Certificate certificate) throws IdentityOA
      */
     public static void validateRequestTenantDomain(String tenantDomainOfApp) throws InvalidOAuthClientException {
 
+        validateRequestTenantDomain(tenantDomainOfApp, null);
+    }
+
+    /**
+     * Validates whether the tenant domain set in context matches with the app's tenant domain in tenant qualified
+     * URL mode.
+     *
+     * @param tenantDomainOfApp Tenant domain of the app.
+     * @param tokenReqDTO       Access token request DTO object that contains request parameters.
+     * @throws InvalidOAuthClientException
+     */
+    public static void validateRequestTenantDomain(String tenantDomainOfApp, OAuth2AccessTokenReqDTO tokenReqDTO)
```

Comment:
better if this can be restructured as follows, 
- call the overloaded method (instead above overloaded method calling this with a null param)
- do the additional work with the tokenReqDTO

Reply:
refactored the code